common/beerocks/bwl/include/bwl/sta_wlan_hal.h

```diff
@@ -48,6 +48,8 @@ class sta_wlan_hal : public virtual base_wlan_hal {
                          bool mem_only_psk, const std::string &bssid, uint8_t channel,
                          bool hidden_ssid) = 0;
 
+    virtual bool associate(const std::string &bssid, uint8_t channel) = 0;
```

Comment:
Please don't use strings for mac addresses in new functions.
```suggestion
    virtual bool associate(const sMacAddr &bssid, uint8_t channel) = 0;
```
Yes, it requires an additional string-to-mac conversion in the caller, and then a mac-to-string conversion to send it to wpa_supplicant, but this is the only way we can ever hope to stamp out the evil use of strings for mac addresses.

Reply:
fixed